pyiceberg/catalog/__init__.py

```diff
@@ -858,6 +860,12 @@ def _update_and_stage_table(
             enforce_validation=current_table is None,
             metadata_location=current_table.metadata_location if current_table else None,
         )
+        io = self._load_file_io(properties=updated_metadata.properties, location=updated_metadata.location)
+
+        # https://github.com/apache/iceberg/blob/f6faa58/core/src/main/java/org/apache/iceberg/CatalogUtil.java#L527
+        # delete old metadata if METADATA_DELETE_AFTER_COMMIT_ENABLED is set to true
+        if current_table is not None:
+            self._delete_old_metadata(io, current_table.metadata, updated_metadata)
```

Comment:
is this the right place for this operation? 
looking at the java implementation, changes are committed to the table before old metadata files are deleted 
https://github.com/apache/iceberg/blob/d935460bbed5da0ac205a67912b70fa682ee84df/core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java#L125-L126

At this point, changes are only applied to the table metadata but is not yet committed

Reply:
the java implementation uses `deleteAfterCommit` 

https://github.com/apache/iceberg/blob/d935460bbed5da0ac205a67912b70fa682ee84df/core/src/main/java/org/apache/iceberg/CatalogUtil.java#L533